pages/Identity/Registry.md

```diff
@@ -0,0 +1,154 @@
+    event DSNPHandle(address addr, uint64 id, string handle);
+
+    /**
+     * @dev Log when a the handle address is changed
+     * @param addr The new address the handle is pointing at
+     * @param id The numerical id for reference 
+     */
+    event DSNPHandleChange(address addr, uint64 id);
+
+    /**
+     * @dev Resolve a handle to a contract address
+     * @param handle The handle to resolve
+     * 
+     * @throws if not found
+     * @returns Address of the contract
+     */
+    function resolveHandleToAddress(string handle) view external returns (address);
+
+    /**
+     * @dev Resolve an id to a contract address
+     * @param id The id to resolve
+     * 
+     * @throws if not found
+     * @returns Address of the contract
+     */
+    function resolveIdToAddress(uint64 id) view external returns (address);
+
+    /**
+     * @dev Resolve a handle to the id
+     * @param handle The handle to resolve
+     * 
+     * @throws if not found
+     * @returns The uint64 id for the handle
+     */
+    function resolveHandleToId(string handle) view external returns (uint64);
+
+    /**
+     * @dev Register a new handle
+     * @param addr Address for the handle to point at
+     * @param handle The handle for the address
+     * 
+     * MUST be called by someone who is authorized on the contract via `EIP 1271(addr).isValidSignature`
+     * MUST emit DSNPHandle
+     */
+    function register(address addr, string handle) external;
+
+    /**
+     * @dev Register a new handle By EIP-712 Signature
+     * @param r ECDSA Signature r value
+     * @param s ECDSA Signature s value
+     * @param v EIP-155 calculated Signature v value
+     * @param addr Address for the handle to point at
+     * @param handle The handle for the address
+     * 
+     * MUST be signed by someone who is authorized on the contract via `EIP 1271(addr).isValidSignature`
+     * MUST emit DSNPHandle
+     */
+    function registerBySignature(bytes32 r, bytes32 s, uint32 v, address addr, string handle) external;
+
+    /**
+     * @dev Alter a handle resolution
+     * @param newAddr New address for the handle to point at
+     * @param handle The handle to alter
+     * 
+     * MUST be called by someone who is authorized on the contract via `EIP 1271(oldAddr).isValidSignature`
+     * MUST emit DSNPHandleOwnershipChange
+     */
+    function setAddr(address newAddr, string handle) external;
+
+    /**
+     * @dev Register a new handle By EIP-712 Signature
+     * @param r ECDSA Signature r value
+     * @param s ECDSA Signature s value
+     * @param v EIP-155 calculated Signature v value
+     * @param newAddr New address for the handle to point at
+     * @param handle The handle to alter
+     * 
+     * MUST be signed by someone who is authorized on the contract via `EIP 1271(oldAddr).isValidSignature`
+     * MUST emit DSNPHandle
+     */
+    function setAddrSignature(bytes32 r, bytes32 s, uint32 v, address newAddr, string handle) external;
```

Comment:
Discussion:
> Should the newAddr be required to "allow" the assignment (instead of just the oldAddr)?

The discussion resolved to say yes, but in some ways this conflicts with the choice of only allowing 1:1 name to identity address. Will discuss more.

Reply:
Still need to choose if the newAddr needs to allow the assignment, however the 1:1 name to contract address require was relaxed as it was irrelevant as one could just set all the delegations the same on different contracts anyway.